Assets/MixedRealityToolkit/Utilities/CameraCache.cs

```diff
@@ -1,49 +1,84 @@
-                    {   // If the cached camera is active, return it
-                        // Otherwise, our playspace may have been disabled
-                        // We'll have to search for the next available
-                        return cachedCamera;
-                    }
+                    return MixedRealityToolkit.CameraSystem.Main;
                 }
 
-                // If the cached camera is null, search for main
-                var mainCamera = Camera.main;
+                // Otherwise, return a default camera
+                return FindOrCreateDefaultMainCamera();
+            }
+        }
 
-                if (mainCamera == null)
-                {   // If no main camera was found, create it now
-                    Debug.LogWarning("No main camera found. The Mixed Reality Toolkit requires at least one camera in the scene. One will be generated now.");
-                    mainCamera = new GameObject("Main Camera", typeof(Camera)) { tag = "MainCamera" }.GetComponent<Camera>();
+        /// <summary>
+        /// Returns true if cached camera exists and is enabled.
+        /// </summary>
+        public static bool MainExists
+        {
+            get
+            {
+                if (MixedRealityToolkit.IsCameraSystemEnabled)
+                {   // The camera system is guaranteed to create a camera.
+                    return true;
                 }
 
-                // Cache the main camera
-                cachedCamera = mainCamera;
+                return cachedCamera != null && cachedCamera.isActiveAndEnabled;
+            }
+        }
 
+        /// <summary>
+        /// Creates a default main camera if our cached camera is null or disabled.
+        /// </summary>
+        /// <returns></returns>
+        private static Camera FindOrCreateDefaultMainCamera()
+        {
+            if (cachedCamera != null && cachedCamera.isActiveAndEnabled)
+            {
+                // If the cached camera is active, return it
+                // Otherwise, our playspace may have been disabled
+                // We'll have to search for the next available
                 return cachedCamera;
             }
+
+            // If the cached camera is null, search for main
+            var mainCamera = Camera.main;
+
+            if (mainCamera == null || !mainCamera.isActiveAndEnabled)
+            {   // If no main camera was found, create it now
+                Debug.LogWarning("No main camera found. The Mixed Reality Toolkit requires at least one camera in the scene. One will be generated now.");
+                mainCamera = new GameObject("Main Camera", typeof(Camera)) { tag = "MainCamera" }.GetComponent<Camera>();
+            }
+
+            // Cache the main camera
+            cachedCamera = mainCamera;
+            // Parent the camera under our play space
+            cachedCamera.transform.SetParent(MixedRealityPlayspace.Transform);
```

Comment:
>MixedRealityPlayspace [](start = 45, length = 21)

We should check whether the playspace exists here, someone may try and use this class without an mrtk configured scene.

Reply:
Calling MixedRealityPlayspace.Transform automatically generates the transform so it's guaranteed to exist.

Reply:
Since this class is static, it reads like it should work independent of mrtk playspaces/components. If there is no camera system, whats the fallout of not creating a playspace/not adding the camera to the playspace? It feels like parenting to a playspace should happen in MixedRealityToolkit.CameraSystem.Main's getter compared to in this class.

---
In reply to: [293890000](https://github.com/microsoft/MixedRealityToolkit-Unity/pull/4455#discussion_r293890000) [](ancestors = 293890000)

Reply:
I agree - IMO the camera cache class should not exist. The camera system should be the only thing responsible for creating or parenting cameras. Creating the Main property on CameraSystem is a baby step in that direction.

Removing CameraCache altogether is beyond the scope of this PR. The goal here is just to ensure that you can additively load multiple cameras / playspace transforms without breaking the system.

We should open a proposal for removing CameraCache so we can discuss impact.

Reply:
Created: #4896